spec/presenters/tree_builder_miq_action_cat_spec.rb

```diff
@@ -0,0 +1,74 @@
+describe TreeBuilderMiqActionCat do
+  before do
+    role = MiqUserRole.find_by_name("EvmRole-operator")
+    @group = FactoryGirl.create(:miq_group, :miq_user_role => role, :description => "Tags Group")
+    login_as FactoryGirl.create(:user, :userid => 'tags_wilma', :miq_groups => [@group])
+    @tag1 = FactoryGirl.create(:classification, :name => 'tag_selected', :show => false)
+    @folder1 = FactoryGirl.create(:classification, :name => 'folder_selected', :show => true)
+    @folder1.entries.push(@tag1)
+    @tag2 = FactoryGirl.create(:classification, :name => 'tag_not_selected', :show => false)
+    @folder2 = FactoryGirl.create(:classification, :name => 'folder_not_selected', :show => true)
+    @folder2.entries.push(@tag2)
+    @group = FactoryGirl.create(:miq_group)
+    @tenant = "TestTenant"
+    @tenant = "#{@tenant} Tags"
+  end
+  context 'read-only mode' do
+    before do
+      @treeName = 'action_tags'
+      @tree = TreeBuilderMiqActionCat.new(@treeName, @treeName, {}, true, @tenant)
+    end
+    it 'set init options correctly' do
+      #binding.pry
+      tree_options = @tree.send(:tree_init_options, @treeName)
+      expect(tree_options).to eq(:expand => true, :lazy => false)
+    end
+    it 'set locals for render correctly' do
+      locals = @tree.send(:set_locals_for_render)
+      expect(locals[:id_prefix]).to eq('cat_tree')
+      expect(locals[:click_url]).to eq("/miq_policy/action_tag_pressed/")
+      expect(locals[:onclick]).to eq("miqOnClickTagCat")
+    end
+    it 'set node' do
+      desc1 = _("Category: %{description}") % {:description => @tag1.description}
+      desc2 = @tag1.description
+      nodeB = TreeNodeBuilderMiqActionCat.new(@tag1, @folder1.id, {})
+      node = nodeB.send(:classification_node)
+
+      expect(node[:hideCheckbox]).to eq true
+      expect(node[:tooltip]).to eq desc1
+      expect(node[:title]).to eq desc2
+
+      desc1 = _("Category: %{description}") % {:description => @folder1.description}
+      desc2 = @folder1.description
+      nodeB = TreeNodeBuilderMiqActionCat.new(@folder1, nil, {})
+      node = nodeB.send(:classification_node)
+
+      expect(node[:hideCheckbox]).to eq true
+      expect(node[:tooltip]).to eq desc1
+      expect(node[:title]).to eq desc2
+      expect(node[:cfmeNoClick]).to eq true
+    end
+
+    it 'sets root' do
+      roots = @tree.send(:root_options)
+      expect(roots).to eq([@tenant, @tenant, "tag.png"])
+    end
+    it 'sets first level nodes correctly' do
+      roots = @tree.send(:x_get_tree_roots, false, nil)
+      expect(roots).to eq([@folder1, @folder2].sort_by! { |c| c.description.downcase })
```

Comment:
Weird to see `sort_by!` when you never use the original array - `sort_by` would be better as it returns a sorted copy instead.